### PR TITLE
Added blog post extensions

### PR DESCRIPTION
**Pull request by g3rb3n**
# Description
Added register_extension and unregister_extension to admin.

## References
implements #569

# Checklist

* [x] I have read the [contribution guide](https://djangocms-blog.readthedocs.io/en/latest/contributing.html)
* [x] Code lint checked via `inv lint`
* [x] ``changes`` file included (see [docs](https://djangocms-blog.readthedocs.io/en/latest/contributing.html#pull-request-guidelines))
* [x] Usage documentation added in case of new features
* [x] Tests added

### CODE REVIEW

File: djangocms_blog/admin.py [hunk 1]
```diff
@@ -19,6 +21,47 @@
 from .models import BlogCategory, Post
 from .settings import get_setting
 
+signal_dict = {}
+
+
+def register_extension(klass):
+    if issubclass(klass, admin.TabularInline):
```
Comment (yakky) [inline djangocms_blog/admin.py:28]: any reason to restrict this to `TabularInline`? Why not `InlineModelAdmin`?

**Comment by yakky** [comment]
@g3rb3n LGTM, it feels a bit weird you have to register a model or an admin, but any other solution I thought of was clumsier, thus I think it's fine.
I just added a minor comment because it's strange to only allow `TabularAdminLine` instead of any inline admin

File: djangocms_blog/admin.py (hunk 1 above)
Comment (g3rb3n) [reply djangocms_blog/admin.py]: Good point.

**Comment by g3rb3n** [comment]
@yakky I have implemented the parent class for the inline. And I have rebased the branch on the latest develop commit.

**Comment by yakky** [comment]
@g3rb3n big fat :+1: 
Thanks a lot for your contribution, it's a huge step

**Pull request merged**